handlers/graph.go

```diff
@@ -181,11 +181,32 @@ func buildNamespaceTrafficMap(namespace string, o options.Options, client *prome
 	// fetch the internally originating request traffic time-series
 	intVector := promQuery(query, time.Unix(o.QueryTime, 0), client.API())
 
+	// 4) query TCP Tx traffic
+	groupByTcp := "source_workload_namespace,source_workload,source_app,source_version,destination_service_namespace,destination_service_name,destination_workload,destination_app,destination_version"
+	query = fmt.Sprintf("sum(rate(%v{reporter=\"source\",source_workload_namespace=\"%v\"} [%vs])) by (%v)",
+		o.TcpTxMetric,
+		namespace,
+		int(o.Duration.Seconds()), // range duration for the query
+		groupByTcp)
+
+	tcpTxVector := promQuery(query, time.Unix(o.QueryTime, 0), client.API())
+
+	// 5) query TCP Rx traffic
+	query = fmt.Sprintf("sum(rate(%v{reporter=\"source\",source_workload_namespace=\"%v\"} [%vs])) by (%v)",
+		o.TcpRxMetric,
+		namespace,
+		int(o.Duration.Seconds()), // range duration for the query
+		groupByTcp)
+
+	tcpRxVector := promQuery(query, time.Unix(o.QueryTime, 0), client.API())
+
```

Comment:
I'd prefer to see all of the HTTP queries in one block.  So, perhaps move the TCP queries down below the istio section and give it it's own comment like:
`// query prometheus for TCP request traffic in two queries:`

Note that if we can get TCP requests from unknown we may need another query, analogous to what we do above for HTTP requests.

I think it is correct to be using source telemetry (reporter="source") for graph building.   Destination telemetry may be appropriate for the side panel metric querying because it is focused on the destination workload.  I'm not 100% convinced we need to gather both sent and received bytes.  We could save a query if we gathered only one, probably received bytes.

Reply:
I removed the tcp_received_bytes query. But also added two other queries: the one for the unknown service, and the one for "external" namespace traffic. I missed those ones. Realized about this when fixing the tests.